presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -323,17 +324,17 @@ public static PlanNode pushPredicateIntoTableScan(
             expression = predicate;
         }
 
-        // optimize rowExpression and return ValuesNode if false. e.g. 0=1 => false, 1>2 => false, (a = 1 and a = 2) => false
-        DomainTranslator.ExtractionResult<VariableReferenceExpression> translatedExpression = translator.fromPredicate(session.toConnectorSession(), expression, BASIC_COLUMN_EXTRACTOR);
-        if (translatedExpression.getTupleDomain().isNone()) {
-            return new ValuesNode(idAllocator.getNextId(), node.getOutputVariables(), ImmutableList.of());
-        }
-
         BiMap<VariableReferenceExpression, VariableReferenceExpression> symbolToColumnMapping = node.getAssignments().entrySet().stream()
                 .collect(toImmutableBiMap(
                         Map.Entry::getKey,
                         entry -> new VariableReferenceExpression(getColumnName(session, metadata, node.getTable(), entry.getValue()), entry.getKey().getType())));
-        PushdownFilterResult pushdownFilterResult = metadata.pushdownFilter(session, node.getTable(), replaceExpression(expression, symbolToColumnMapping));
+
+        RowExpression replacedExpression = replaceExpression(expression, symbolToColumnMapping);
+        // replaceExpression() may further optimize the expression, if the resulting expression is false and return an empty result
```

Comment:
typos: `replaceExpression() may further optimize the expression; if the resulting expression is always false, then return empty Values node`